public/components/mainMenu.js

```diff
@@ -0,0 +1,110 @@
+const { dialog } = require('electron');
+const updater = require('./updater');
+
+const openDialogOptions = {
+  buttonLabel: 'Open',
+  nameFieldLabel: 'Open:',
+  defaultPath: 'Protocol.netcanvas',
+  filters: [{ name: 'Network Canvas', extensions: ['netcanvas'] }],
+  properties: ['openFile'],
+};
+
+const openDialog = () =>
+  new Promise((resolve, reject) => {
+    dialog.showOpenDialog(openDialogOptions, (filename) => {
+      if (filename === undefined) { reject(); return; }
+      resolve(filename[0]);
+    });
+  });
+
+const openFile = window =>
+  () =>
+    openDialog()
+      .then(filePath => window.webContents.send('OPEN_FILE', filePath));
+
+const MenuTemplate = (window) => {
+  const menu = [
+    {
+      label: 'File',
+      submenu: [
+        {
+          label: 'Open...',
+          click: openFile(window),
+        },
+      ],
+    },
+    {
+      label: 'Edit',
+      submenu: [
+        { role: 'cut' },
+        { role: 'copy' },
+        { role: 'paste' },
+        { type: 'separator' },
+        { role: 'selectall' },
+      ],
+    },
+    {
+      label: 'View',
+      submenu: [
+        { role: 'resetzoom' },
```

Comment:
We should consider removing all of the zoom controls since text/content is no longer zoomable in the app (instead being tied to viewport dimensions).

Reply:
I'm going to leave this for now, on the off chance that someone has zoomed the app in, and this causes issues with them not being able to reset it. I created an issue to update this menu item to use a new system in the meaintime.